### PR TITLE
[CVAT] Add polygons support

### PR DESCRIPTION
**Pull request by Bobronium**
## Description
Add initial Polygons support to Exchange and Recording Oracles.
Validation is implemented through converting polygons to bounding boxes and using existing `bbox_iou` function. 
This is not accurate by any means, but should be enough until we move validation to CVAT.

## Summary of changes
- Add required types for polygons support

## How test the changes
Considering you're running `docker-compose.dev.yml`:
- Upload [cheemsdoge_manifest.json](https://github.com/user-attachments/files/17633307/cheemsdoge_manifest.json) to http://127.0.0.1:9010/manifests/cheemsdoge_manifest.json
- Upload [cheemsdoge_gt.json](https://github.com/user-attachments/files/17633305/cheemsdoge_gt.json) to http://127.0.0.1:9010/launcher
- And upload these to http://127.0.0.1:9010/launcher/polygons-data:
	- [cheems.jpg](https://github.com/user-attachments/assets/9d2cf3b6-bf5d-430c-a1a4-f1924035a869)
	- [cheemsdoge_gt.jpg](https://github.com/user-attachments/assets/a4696e96-d06a-485b-9a97-177ba16e0b07)

Then, go through the usual process of creating, annotating and submitting annotations. You should see annotations results uploaded to http://127.0.0.1:9010/recor-data/:9010/cheemsdoge_manifest.json@80002/resulting_annotations.zip

### CODE REVIEW

File: packages/examples/cvat/recording-oracle/src/handlers/process_intermediate_results.py [hunk 1]
```diff
@@ -60,6 +62,7 @@
 DATASET_COMPARATOR_TYPE_MAP: dict[TaskTypes, type[DatasetComparator]] = {
     # TaskType.image_label_binary: TagDatasetComparator, # TODO: implement if support is needed
     TaskTypes.image_boxes: BboxDatasetComparator,
+    TaskTypes.image_polygons: BboxDatasetComparator,
```
Comment (zhiltsov-max) [inline packages/examples/cvat/recording-oracle/src/handlers/process_intermediate_results.py:65]: Probably, a different class is needed here.

File: packages/examples/cvat/exchange-oracle/src/handlers/completed_escrows.py [hunk 2]
```diff
@@ -282,3 +282,4 @@ def handle_escrows_validations(logger: logging.Logger) -> None:
                 chain_id=chain_id,
                 status=EscrowValidationStatuses.in_progress,
             )
+            session.commit()
```
Comment (zhiltsov-max) [inline packages/examples/cvat/exchange-oracle/src/handlers/completed_escrows.py:285]: Doesn't session commit automatically?

Comment (Bobronium) [reply packages/examples/cvat/exchange-oracle/src/handlers/completed_escrows.py]: You're right, it does. Leftovers from debugging, removed.

**Review by zhiltsov-max — COMMENTED** [review]
Please extract the dev patches into another PR to review and merge it separately. The part with polygons seems to be OK, taking into account the validation simplification discussed.

File: packages/examples/cvat/exchange-oracle/.gitignore [hunk 3]
```diff
@@ -6,3 +6,5 @@ __pycache__/
 .env
 .python-version
 
+human_app_public_key.pem
+human_app_private_key.pem
```
Comment (zhiltsov-max) [inline packages/examples/cvat/exchange-oracle/.gitignore:10]:
```suggestion
```

Comment (Bobronium) [reply packages/examples/cvat/exchange-oracle/.gitignore]: Good catch, thanks!

**Review by zhiltsov-max — APPROVED** [review]

**Review by Dzeranov — APPROVED** [review]

**Pull request merged**